### DISCUSSION TITLE
What are the ramifications of changing the module name in go.mod?

### DISCUSSION THREAD

**Discussion by shueybubbles** [microsoft/go-mssqldb#5] [Q&A]
This is my first hard fork of a go module so I'm not sure what the right procedure is regarding the module name in go.mod.
Should we keep it as `github.com/denisenkom/go-mssqldb` until we get V1 done and rename v2 to `github.com/microsoft/go-mssqldb` then? 
Should we rename it now?
Or should we just keep the name as-is forever?
I am looking for supporting arguments for each option.
thx!

**Comment by krowbar** [comment]
I believe you need to change the the name in go.mod to make it fully functional as a go module. In a simple test where I try to get a new project started:

`go mod tidy`
`go: finding module for package github.com/microsoft/go-mssqldb`
`go: found github.com/microsoft/go-mssqldb in github.com/microsoft/go-mssqldb v0.12.0`
`go: rkruus/ms_new imports`
`        github.com/microsoft/go-mssqldb: github.com/microsoft/go-mssqldb@v0.12.0: parsing go.mod:`
 `       module declares its path as: github.com/denisenkom/go-mssqldb`
  `              but was required as: github.com/microsoft/go-mssqldb`


or:

`go get github.com/microsoft/go-mssqldb: `
`go: github.com/microsoft/go-mssqldb@v0.12.0: parsing go.mod:`
`        module declares its path as: github.com/denisenkom/go-mssqldb`
`                but was required as: github.com/microsoft/go-mssqldb`

**Reply by shueybubbles** [reply]
thx that makes sense as what we aspire to have eventually.
I am interested in whether the community would prefer us to make that name change now or if they'd rather use a `replace` directive in their apps to redirect `denisenkom` references to `microsoft`. There are also documentation and supportability concerns - lots of docs reference `denisenkom` today and over time we want to have those sites reference this fork. Once we rename the module, how do we get it to show up in go docs etc? Point me to relevant sources and I'll go from there.

**Reply by krowbar** [reply]
https://pkg.go.dev/about#adding-a-package 
Not sure how you can get redirects to here -- ideally a note  on the `denisenkom` side would work best (and a note in the godocs on pkd.go.dev as well).
I assume the links in the README.md here would have to be updated as well.

Right now using `replace` does not work either as it does not see you repository:
`go mod edit -replace github.com/denisenkom/go-mssqldb@v0.12.0=github.com/microsoft/go-mssqldb@v0.12.0`
`go get -u`
`github.com/microsoft/go-mssqldb: cannot find module providing package github.com/microsoft/go-mssqldb`
or just pulls in the `denisemkom` version and calls it `microsoft` (the queries_go19_test.go are the same in my go/pkg/mod directories for those versions)

**Comment by krowbar** [comment]
Looks like https://github.com/vcabbage/amqp/ had the same thing happen (see https://github.com/vcabbage/amqp/issues/205).
The go reference link from the old vcabbage repo points to the new package documentation -- https://pkg.go.dev/github.com/Azure/go-amqp.
The Azure repo has updated the go.mod with their project name.

**Reply by shueybubbles** [reply]
I have a PR at #12 to do it.
